yt_dlp/extractor/substack.py

```diff
@@ -0,0 +1,98 @@
+            'thumbnail': 'md5:daa40b6b79249417c14ff8103db29639',
+            'uploader': 'haleynahman',
+        }
+    }, {
+        'url': 'https://andrewzimmern.substack.com/p/mussels-with-black-bean-sauce-recipe',
+        'md5': 'fd3c07077b02444ff0130715b5f632bb',
+        'info_dict': {
+            'id': '47368578',
+            'ext': 'mp4',
+            'title': 'Mussels with Black Bean Sauce: Recipe of the Week #7',
+            'description': 'md5:b96234a2906c7d854d5229818d889515',
+            'thumbnail': 'md5:e30bfaa9da40e82aa62354263a9dd232',
+            'uploader': 'andrewzimmern',
+        }
+    }]
+
+    @staticmethod
+    def is_custom_domain(webpage):
+        return re.search(r'<script[^>]+src=["\']https://substackcdn.com/.*\.js', webpage) is not None
+
+    @staticmethod
+    def url_from_custom_domain(url):
+        custom_domain = r'https?://(?:www\.)?(?P<username>[\w-]+)\.(?:[a-z]+)/(?P<params>.+)'
+
+        username = re.search(custom_domain, url).group('username')
+        params = re.search(custom_domain, url).group('params')
+
+        return f'https://{username}.substack.com/{params}'
+
+    def _extract_video_formats(self, video_id, username):
+        formats, subtitles = [], {}
+        for video_format in ('hls', 'mp4'):
+            video_url = f'https://{username}.substack.com/api/v1/video/upload/{video_id}/src?type={video_format}'
+
+            if video_format == 'hls':
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(video_url, video_id, 'mp4', fatal=False)
+                formats.extend(fmts)
+                self._merge_subtitles(subs, target=subtitles)
+            else:
+                formats.append({
+                    'url': video_url,
+                    'ext': video_format,
+                })
+
+        return formats, subtitles
+
+    def _real_extract(self, url):
+        display_id, username = self._match_valid_url(url).group('id', 'username')
+        webpage = self._download_webpage(url, display_id)
+
+        post_info = self._search_json(
+            r'<script[^>]*>\s*window\._preloads\s*=\s*', webpage, 'preloads', display_id)['post']
+
+        post_type = post_info.get('type')
+        formats, subtitles = [], {}
+        if post_type == 'podcast':
+            formats, subtitles = [{'url': post_info['podcast_url']}], {}
+        elif post_type == 'video':
+            formats, subtitles = self._extract_video_formats(post_info['videoUpload']['id'], username)
+        else:
+            self.raise_no_formats(f'Page type "{post_type}" is not supported')
+
+        self._sort_formats(formats)
+        return {
+            'id': str(post_info['id']),
+            'formats': formats,
+            'subtitles': subtitles,
+            'title': post_info.get('title'),
+            'description': post_info.get('description'),
+            'thumbnail': post_info.get('cover_image'),
+            'uploader': username,
```

Comment:
`uploader` should probably somehow point to the Substack profile page (under `https://substack.com/profile/`). What you call `username` here is actually the name of the newsletter only.

Compare mollymovieclub.com and computerenhance.substack.com being both written by the same author.

Reply:
I think uploader refers to the channel/newsletter/blog where it was uploaded, not the author 🤔 

I've improved it a bit in https://github.com/yt-dlp/yt-dlp/pull/4011/commits/f9ab764f2530cd6ea44189cfd02da855d94af24e, what do you think?